### PR TITLE
DolphinQt/Mapping: Add "Use Mouse Controlled Pointing" button.

### PR DESCRIPTION
**Pull request by luc-git**
This PR adds a checkbox which when it is checked it maps mouse directions to the IR cursor.
One problem I didn't find how to save this state in the ini config. So when you close and restart dolphin it will disable checkbox even if it's enabled

### CODE REVIEW

**Comment by JosJuice** [comment]
So this effectively resets the Point mappings to their defaults, without touching the other mappings? If we are to have that as an option, I think it would make more sense as a button than as a checkbox.

**Comment by jordan-woyak** [comment]
It sounds like you're trying to make a convenient way to bind the typical mouse controlled pointing without having to click the "Default" button that loads literally *all* of the keyboard/mouse defaults. I agree that this would be useful, but I don't think it makes sense as a checkbox. I think it would make more sense as a button, or maybe as an item to pick in the "Refresh" drop down box. I don't think "Use Mouse as IR" is a good name. "Load Default Mouse-Controlled Pointing" or similar, maybe?

**Comment by luc-git** [comment]
Any news?

**Comment by mbc07** [comment]
@dolphin-emu-bot rebuild

**Review by mbc07 — CHANGES_REQUESTED** [review]
I'm not particularly thrilled in selectively adding "reset to defaults" button to specific subsets of the mappings and can already picture things getting out of hand fast, but if we're really going ahead with this one, I'd like to see the following points addressed before giving approval:

* The added "Use Mouse Controlled Pointing" is all the way down, after a few bindings it doesn't affect at all. This looks odd to me, so I'd like to see that button moved to the top, right below the existing "Calibrate" button, thus, more close to the affected bindings.
* The default mappings that are applied when the button is pressed are hardcoded, and to different defaults than we currently have (prefixed ":" before the binding expression). This looks like a potential code smell and, although it would require more changes, I think `m_controller->LoadDefaults()` should be extended instead to be able to load the defaults only for a specific binding group and be called there instead.
* In its current state, this PR is a 20 LOC change, having it spread across 7 separate commits (with reverts alongside) seems very excessive. Please squash that to a single commit.

**Comment by mbc07** [comment]
@dolphin-emu-bot rebuild

**Review by mbc07 — COMMENTED** [review]
I still think this should be addressed:
> * The default mappings that are applied when the button is pressed are hardcoded, and to different defaults than we currently have (prefixed ":" before the binding expression). This looks like a potential code smell and, although it would require more changes, I think `m_controller->LoadDefaults()` should be extended instead to be able to load the defaults only for a specific binding group and be called there instead.

Other than that, this LGTM

**Comment by luc-git** [comment]
> I still think this should be addressed:
> > * The default mappings that are applied when the button is pressed are hardcoded, and to different defaults than we currently have (prefixed ":" before the binding expression). This looks like a potential code smell and, although it would require more changes, I think `m_controller->LoadDefaults()` should be extended instead to be able to load the defaults only for a specific binding group and be called there instead.
> 
> Other than that, this LGTM

But if you set other controller than the mouse as default the expression is also prefixed by ":"

**Comment by AdmiralCurtiss** [comment]
I'm guessing here, so correct me if I'm wrong, but,

I'm getting the impression here that you want to be able to use your Mouse as a the Point device even if you have a different device (like your Controller) selected as your default mapping device, and that's why you put the `:` in the prefix there because that's how it shows up in the box if you map like that manually. That will not actually work by just putting a `:` into the actual expression, the `:` shown is just a shorthand for 'explicit device selected'.

In order to make that work, you would need to find the default Mouse device and prefix its actual device identifier in front of the expression. This will differ depending on your hardware and OS, so you can't just hardcode the string; presumably you need to ask the `g_controller_interface` to give you the correct device.

**Comment by luc-git** [comment]
> I'm guessing here, so correct me if I'm wrong, but,
> 
> ~I'm getting the impression here that you want to be able to use your Mouse as a the Point device even if you have a different device (like your Controller) selected as your default mapping device, and that's why you put the `:` in the prefix there because that's how it shows up in the box if you map like that manually. That will not actually work by just putting a `:` into the actual expression, the `:` shown is just a shorthand for 'explicit device selected'.~
> 
> ~In order to make that work, you would need to find the default Mouse device and prefix its actual device identifier in front of the expression. This will differ depending on your hardware and OS, so you can't just hardcode the string; presumably you need to ask the `g_controller_interface` to give you the correct device.~
> 
> e: Oh wait, you're already doing that! I didn't even recognize that in the code until I thought about this myself due to the way you're inserting the expression into the string -- I'd recommend just using `fmt::format()` instead so it's more obvious what you're doing.
> 
> Relying on the default device being the Mouse and having Cursor mappings is also a bit sketchy, but should be correct on all Desktop OSes which is all that Qt is ever going to be used on, so I'm not sure if it makes sense to make this more complicated just to verify that.

It's done the code uses `fmt::format()` now

**Comment by mbc07** [comment]
@dolphin-emu-bot rebuild

**Review by mbc07 — APPROVED** [review]
I think I finally understand what you're trying to do with the bindings now. Not exactly how I thought that last point I was talking would be addressed, but it works. 

LGTM after you squash the commits into a single one...

**Comment by luc-git** [comment]
@JosJuice Is it good?

File: Source/Core/DolphinQt/Config/Mapping/MappingWidget.cpp
Comment (JosJuice) [inline Source/Core/DolphinQt/Config/Mapping/MappingWidget.cpp:169]: I don't think you need `mouse_button` in the capture list.

**Review by JosJuice — COMMENTED** [review]
Well, personally I don't know if we need this feature, but as far as the implementation is concerned I only have one thing to comment on:

**Comment by luc-git** [comment]
> Well, personally I don't know if we need this feature, but as far as the implementation is concerned I only have one thing to comment on:

Fixed

**Comment by AdmiralCurtiss** [comment]
Well, I think a feature in general to only reset a subset of controls to a known default makes sense, though I personally would like to also see it for things like gyro mappings and other things that are common but annoying to map manually -- at which point a more generic implementation should probably replace this. I suppose this is fine for now though.

**Pull request merged**
